tests/tests/rpc_module.rs

```diff
@@ -31,6 +31,23 @@ use jsonrpsee::core::Error;
 use jsonrpsee::types::{EmptyParams, Params};
 use serde::{Deserialize, Serialize};
 
+fn assert_type<T: 'static, Expected: 'static>(_expected: &Expected, explain: Option<&'static str>) {
+	use core::any::TypeId;
+	match explain {
+		None => assert_eq!(TypeId::of::<T>(), TypeId::of::<Expected>()),
+		Some(explain) => assert_eq!(TypeId::of::<T>(), TypeId::of::<Expected>(), "{}", explain),
+	}
+}
```

Comment:
Perhaps it's better to move this check to compile-time? ie:

```rust
fn assert_type<Expected: 'static>(_expected: &Expected)
```

Usage (compile error if types don't match):

```
assert_type::<SomeType>(some_value);
```

Reply:
cheers!